### ISSUE TITLE
Extract names and scores of packages

### ISSUE THREAD

**Issue by jayoung-lee** [dart-lang/pub-dartlang-dart#1537]
I would like to extract the names and scores of Flutter packages from [https://pub.dartlang.org/flutter/packages](https://pub.dartlang.org/flutter/packages) and track the number of quality packages (i.e., score > TBD) over time. 

Is it possible to export this data to a file system (e.g., Colossus) and update it periodically?

**Comment by isoos** [comment]
At this point I would recommend against it, because score is not exact, and often changes because:
- we change how scoring works (or just fix bugs)
- the analysis SDKs change (and they report different issues)
- the package releases a new version

It would not be meaningful to compare current scores with scores 6 months ago.

**Comment by jayoung-lee** [comment]
Hi István! 

The purpose is to keep track of the growth of Flutter ecosystem. We can start from the number of packages. 

Yet, we will eventually take the quality of packages into account. Can I expect that the scoring system will be stabilized someday? I can start tracking the quality once the scoring system is ready and meaningful to use, or use sub-scores that are relatively more stable. 

Regarding the release of a new version, would the score be reset to 50?

**Comment by isoos** [comment]
@jayoung-lee: with today's release you can see:
- the [`uses-flutter`](https://pub.dartlang.org/api/packages/shared_preferences/metrics?pretty) flag, and
- the `overallScore` in the output, and
- also have access to the details of the analysis reports, with the extra query parameter:
https://pub.dartlang.org/api/packages/pana/metrics?reports
(also: you can use the [&pretty](https://pub.dartlang.org/api/packages/pana/metrics?reports&pretty) formatting for human-readable indents).

I'm not aware of anything else that may be missing from this, please open a new issue if there is a need for some kind of data that we don't expose.
